e2e/etcdctlv3_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coreos/etcd/pkg/fileutil"
+	"github.com/coreos/etcd/pkg/testutil"
+)
+
+func TestCtlV3SetQuorum(t *testing.T) {
+	testCtlV3Set(t, &configNoTLS, 5*time.Second, true)
+}
+
+func TestCtlV3SetQuorumZeroTimeout(t *testing.T) {
+	testCtlV3Set(t, &configNoTLS, 0, true)
+}
+
+func TestCtlV3SetQuorumTimeout(t *testing.T) {
+	testCtlV3Set(t, &configNoTLS, time.Nanosecond, true)
+}
+
+func TestCtlV3SetPeerTLSQuorum(t *testing.T) {
+	testCtlV3Set(t, &configPeerTLS, 5*time.Second, true)
+}
+
+func testCtlV3Set(t *testing.T, cfg *etcdProcessClusterConfig, dialTimeout time.Duration, quorum bool) {
+	defer testutil.AfterTest(t)
+
+	epc := setupEtcdctlV3Test(t, cfg, quorum)
+	defer func() {
+		if errC := epc.Close(); errC != nil {
+			t.Fatalf("error closing etcd processes (%v)", errC)
+		}
+	}()
+
+	key, value := "foo", "bar"
+
+	donec := make(chan struct{})
+	go func() {
+		if err := etcdctlV3Put(epc, key, value, dialTimeout); err != nil {
+			if dialTimeout > 0 && dialTimeout <= time.Nanosecond && isGRPCTimedout(err) { // timeout expected
+				donec <- struct{}{}
+				return
+			}
+			t.Fatalf("put error (%v)", err)
+		}
+		if err := etcdctlV3Get(epc, key, value, dialTimeout, quorum); err != nil {
+			if dialTimeout > 0 && dialTimeout <= time.Nanosecond && isGRPCTimedout(err) { // timeout expected
+				donec <- struct{}{}
+				return
+			}
+			t.Fatalf("get error (%v)", err)
+		}
+		donec <- struct{}{}
+	}()
+
+	select {
+	case <-time.After(2*dialTimeout + time.Second):
+		if dialTimeout > 0 {
+			t.Fatalf("test timed out for %v", dialTimeout)
+		}
+	case <-donec:
+	}
+}
+
+func etcdctlV3PrefixArgs(clus *etcdProcessCluster, dialTimeout time.Duration) []string {
```

Comment:
can we remove all etcd mentioning in this file? it is clearly etcd specific test.